avar2.md

```diff
@@ -30,27 +30,36 @@ Use cases include:
 
 The table format for `avar` version 2 is the same as `avar` version 1, appended with offsets to two extra structures, `axisIndexMap` and `varStore`, and with the data for those structures.
 
-`axisIndexMap` is a *DeltaSetIndexMap* structure that maps the axis indices implied in `fvar` to indices used in `varStore`. The outer index identifies an *ItemVariationData* structure in `varStore`. The inner index identifies a *deltaSet* within an *ItemVariationData*.
+`axisIndexMap` is a *DeltaSetIndexMap* structure that maps the axis indices implied in `fvar` to indices used in `varStore`. In the mapping, the index of each mapping entry is the axis index; the outer index identifies an *ItemVariationData* structure in `varStore`; the inner index identifies a *deltaSet* within an *ItemVariationData*. If either outer or inner index is 0xffff, then that axis’s value is ignored by the `avar` version 2 process. The number of mapping entries may be smaller than the number of axes in 'fvar', in which case remaining axes are ignored.
```

Comment:
> If either outer or inner index is 0xffff, then that axis’s value is ignored by the `avar` version 2 process.

The OpenType spec only says the pair 0xFFFF,0xFFFF means "no variation".

> The number of mapping entries may be smaller than the number of axes in 'fvar', in which case remaining axes are ignored.

OpenType spec says any missing values at the end get the value of the last entry. Again, this is details specified in *DeltaSetIndexMap* structure already.

Reply:
And because of how *DeltaSetIndexMap* compression works, it is more efficient to encode an entry in *ItemVariationStore* for "no variation" and use that, instead of using 0xFFFF,0xFFFF, which would force *DeltaSetIndexMap* entries to 4bytes per entry.

Then again, these are compiler-optimization issues, which IMO should NOT be part of the spec. Definitely not here.

Reply:
Ah indeed it does. (I was looking for such an explanation in the docs for *DeltaSetIndexMap* when in fact it applies whatever method is used to find outer and inner.)

Reply:
I saw in your RobotoFlex out.ttf demo font you had used 6 *ItemVariationData*s. The reason to do that is to avoid proliferation of zeroes in the *deltaSet*s, isn’t it?

Reply:
Not just zeroes. Also to choose the optimal encoding using one-byte vs two-byte columns. We have an optimizer that takes care of it.

Reply:
Got it